services/iaas/wait/wait.go

```diff
@@ -110,6 +110,27 @@ func CreateNetworkWaitHandler(ctx context.Context, a APIClientInterface, project
 	return handler
 }
 
+// CreateNetworkWaitHandlerWithNetworkId will wait for network creation using network id
+func CreateNetworkWaitHandlerWithNetworkId(ctx context.Context, a APIClientInterface, projectId, networkId string) *wait.AsyncActionHandler[iaas.Network] {
```

Comment:
I think we could already replace the old waiter as it is not needed anymore, and then we can replace it in the TFP

Reply:
Renaming this back to the correct name afterwards would be a breaking change anyway